docs/src/references/mqtt-topic-guidelines.md

```diff
@@ -0,0 +1,242 @@
+   can be understood from the topic directly.
+   For e.g: a topic scheme like `tedge/main/{id}`, `tedge/service/{id}` and `tedge/child/{id}` is more user-friendly
+   than a simpler context agnostic scheme like `tedge/{id}` where `id` can be for any "thing".
+1. Allowing the "things" in a local domain to use their user-friendly unique IDs (within the tedge namespace)
+   in MQTT topics over their globally unique cloud IDs (typically very long and cumbersome) would be desired.
+   Ideally, all translation between user-provided-ids and cloud-twin-ids should be done internally by thin-edge,
+   as long as it doesn't adversely affect any data mapping cost at scale.
+1. Limit the topic levels to 7 as AWS IoT core has a [max limit of 7](https://docs.aws.amazon.com/whitepapers/latest/designing-mqtt-topics-aws-iot-core/mqtt-design-best-practices.html)
+1. Symmetric topics: It would be good to keep the subtopic levels symmetric so that the wildcard subscriptions like
+   "subscribe to all measurements from all devices and services" are easier.
+### Out of scope
+
+1. Routing different kinds of data to different clouds, e.g: all telemetry to azure and all commands from/to Cumulocity.
+   Even though this requirement is realistic, thin-edge MQTT topics must not be corrupted with cloud specific semantics,
+   and the same requirement will be handled with some external routing mechanism(e.g: routing via bridge topics)
+1. Services on child devices. Service support can be limited to thin-edge devices only.
+
+## Proposals
+
+### Dedicated topics for tedge device, services and child devices
+
+The topics for the thin-edge device, the services running on it and child devices have different prefixes:
+
+For parent: tedge/main/<device-id>
+For services: tedge/service/<service-id>
+For immediate child devices: tedge/child/<child-id>
+For nested child devices: tedge/descendent/<child-id>
+
+#### Telemetry
+
+For telemetry data, the topics would be grouped under a `telemetry/` sub-topic as follows:
+* Measurements: `tedge/main/<device-id>/telemetry/measurements`
+* Events: `tedge/main/<device-id>/telemetry/events/<event-type>`
+* Alarms: `tedge/main/<device-id>/telemetry/alarms/<alarm-type>/<severity>`
+
+For child devices and services, a similar structure is followed like: `tedge/child/<child-id>/telemetry/measurements`,
+`tedge/service/<service-id>/telemetry/events/<event-type>` etc
+
+The subtopic levels `main/<device-id>` are really not required for the main device.
+They are added just for the sake of consistency with child devices and services so that
+it is easier to make queries like "subscribe to all measurements from all devices and services"
+which can be achieved with a wildcard subscription on `tedge/+/<device-id>/telemetry/measurements`.
+If this is not desired, then we can simplify it to just `tedge/telemetry/measurements`.
+
+#### Commands
+
+Similarly, all commands would be grouped under a `commands/` sub-topic as follows:
```

Comment:
The inclusion of the `device-id` for the `tedge/main` topic is not required (and not desired) as it makes it harder to publish to commonly used topics without knowning the device-id that you are currently running under.